pandas/core/groupby/generic.py

```diff
@@ -321,13 +321,18 @@ def _aggregate_multiple_funcs(self, arg):
                 obj = copy.copy(obj)
                 obj._reset_cache()
                 obj._selection = name
-            results[name] = obj.aggregate(func)
+            results[base.OutputKey(label=name, position=idx)] = obj.aggregate(func)
 
         if any(isinstance(x, DataFrame) for x in results.values()):
             # let higher level handle
-            return results
+            return {key.label: value for key, value in results.items()}
```

Comment:
Do you know the code path that hits this? Wouldn't this run into the same error if `key.label` is non-unique?

Reply:
This happens if one of the aggregations is 'ohlc' - will look into this, thanks